OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs

```diff
@@ -85,11 +84,28 @@ public void Tick(Actor self)
 				if (lastKnownActorIds.Contains(actor.Actor.ActorID))
 					continue;
 
+				var notificationPlayed = playedNotifications.Contains(actor.Trait.Info.Notification);
+
+				// Notify the actor that he's been discovered
+				foreach (var trait in actor.Actor.TraitsImplementing<INotifyDiscovered>())
+					trait.OnDiscovered(actor.Actor, self.Owner, !notificationPlayed);
```

Comment:
Because that check is only for audio notifications (so they don't spam 10 per seconds and you can't hear anything).
This also calls the script triggers, which I think should be called as soon as possible.